pages/index.tsx

```diff
@@ -141,48 +141,57 @@ export default function Home() {
               <div
                 key={`${giftItem.uuid}_divbutton`}
-                className="animate-width whitespace-nowrap overflow-hidden"
+                className="animate-opacity"
               >
-                <li key={giftItem.uuid}>
+                <li key={giftItem.uuid} className="[overflow-wrap:anywhere]">
                   {giftItem.receiver} - {giftItem.gift}
-                  <Button
-                    onMouseOver={(e) =>
-                      e.currentTarget.parentElement?.setAttribute(
-                        'class',
-                        'line-through',
-                      )
-                    }
-                    onMouseOut={(e) =>
-                      e.currentTarget.parentElement?.removeAttribute('class')
-                    }
-                    key={`${giftItem.uuid}_deletebutton`}
-                    className="ms-5 p-0 w-16 h-8 hover:text-red-600 pointer-events-auto"
-                    onClick={() => {
-                      setDeleteModalGiftData(giftItem);
-                      setIsDeleteModalOpen(true);
-                    }}
-                    type="button"
-                  >
-                    Poista
-                  </Button>
-                  <Button
-                    key={`${giftItem.uuid}_editbutton`}
-                    className="ms-3 p-0 w-20 h-8 hover:text-yellow-400"
-                    onClick={() => {
-                      setEditModalGiftData(giftItem);
-                      setIsEditModalOpen(true);
-                    }}
-                    type="button"
-                  >
-                    Muokkaa
-                  </Button>
                 </li>
+                <Button
+                  onMouseOver={(e) =>
+                    e.currentTarget.parentElement?.setAttribute(
+                      'class',
+                      'line-through',
+                    )
+                  }
+                  onMouseOut={(e) =>
+                    e.currentTarget.parentElement?.removeAttribute('class')
+                  }
+                  key={`${giftItem.uuid}_deletebutton`}
+                  className="m-3 p-0 w-16 h-8 hover:text-red-600 pointer-events-auto"
+                  onClick={() => {
+                    setDeleteModalGiftData(giftItem);
+                    setIsDeleteModalOpen(true);
+                  }}
+                  type="button"
+                >
+                  Poista
+                </Button>
+                <Button
+                  key={`${giftItem.uuid}_editbutton`}
+                  onMouseOver={(e) =>
+                    e.currentTarget.parentElement?.setAttribute(
+                      'class',
+                      'underline',
+                    )
+                  }
+                  onMouseOut={(e) =>
+                    e.currentTarget.parentElement?.removeAttribute('class')
+                  }
+                  className="m-3 ml-0 p-0 w-20 h-8 hover:text-yellow-400"
```

Comment:
Näitä `onMouseOver` ja `onMouseOut` tarvitsee todella harvoin. Itse en muista koskaan käyttäneeni. Kun eikö tämänkin voi korvata `hover:underline` luokalla buttoniin?
```suggestion
                  className="m-3 ml-0 p-0 w-20 h-8 hover:text-yellow-400 hover:underline"
```

Reply:
> Näitä `onMouseOver` ja `onMouseOut` tarvitsee todella harvoin. Itse en muista koskaan käyttäneeni. Kun eikö tämänkin voi korvata `hover:underline` luokalla buttoniin?

Tämän tarkoitus on ala- tai yliviivata lahja riippuen onko kursori `Muokkaa`-  vai  `Poista`-napin päällä.

Lyhykäisyydessään tämä asettaa lahjan div-wrapperille `line-through`- tai `underline`-arvon

```ts
className="m-3 ml-0 p-0 w-20 h-8 hover:text-yellow-400 hover:underline"
```
Tällä tavalla yritin itsekin ratkaista, mutta tämä tapa alleviivaa pelkästään tuon nappulan tekstin

Olisiko tässä kohtaa järkevin antaa tuon `onMouseOver` & `onMouseOut` -tavan olla vai kannattaako se poistaa?

Reply:
Oho totta, en katsonut tarpeeksi tarkasti. Vähän innostuin selvittämään tuota että saako tehtyä pelkällä CSS:llä vai onko tuo sun tapa ainoa. Luulen että sain toimivan CSS-version joka ei tosin ollut ihan simppeli. Pushasin siitä commitin, tsekkaapa toimiiko kuten pitää.